src/main/java/subway/section/Sections.java

```diff
@@ -0,0 +1,116 @@
+package subway.section;
+
+import subway.exception.SubwayException;
+import subway.station.Station;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.List;
+
+@Embeddable
+public class Sections {
+
+    @OneToMany(mappedBy = "line", cascade = CascadeType.PERSIST, orphanRemoval = true)
+    List<Section> sections = new ArrayList<>();
+
+    public Sections() {
+    }
+
+    public void addSection(Section section) {
+        if (sections.size() > 0) {
+            validateNextSection(section);
+            validateDuplicateStation(section);
+        }
+        this.sections.add(section);
+    }
+
+    private void validateNextSection(Section section) {
+        if (!getEndStation().isEquals(section.getUpStation())) {
+            throw new SubwayException("구간의 상행역은 해당 노선에 등록되어있는 하행 종점역이 아닙니다.");
+        }
+    }
+
+    private void validateDuplicateStation(Section section) {
+        if (isContains(section.getDownStation())) {
+            throw new SubwayException("이미 등록되어있는 역입니다.");
+        }
+    }
+
+    private boolean isContains(Station station) {
+        return this.sections.stream().anyMatch(section -> section.getUpStation().equals(station));
+    }
+
+    public void removeSection(Long stationId) {
+        validateLastSection();
+        validateEndSection(stationId);
+
+        Section deleteSection = this.sections.stream()
+                .filter(section -> section.getDownStation().isEquals(stationId))
+                .findFirst()
+                .orElseThrow(() -> new SubwayException("역을 찾을 수 없습니다."));
+
+        this.sections.remove(deleteSection);
+    }
+
+    private void validateLastSection() {
+        if (sections.size() < 2) {
+            throw new SubwayException("구간이 1개인 경우 역을 삭제할 수 없습니다.");
+        }
+    }
+
+    private void validateEndSection(Long stationId) {
+        if (!getEndStation().isEquals(stationId)) {
+            throw new SubwayException("마지막 구간만 제거할 수 있습니다.");
+        }
+    }
+
+    public List<Station> getOrderedStations() {
+        List<Station> stations = new ArrayList<>();
+        Station upStation = getStartStation();
+        stations.add(upStation);
+
+        for (int i = 0; i < sections.size(); i++) {
+            upStation = findNextStation(upStation);
+            stations.add(upStation);
+        }
+
+        return stations;
+    }
+
+    private Station findNextStation(Station upStation) {
+        return sections.stream()
+                .filter(section -> section.equalsUpStation(upStation))
+                .map(Section::getDownStation)
+                .findFirst()
+                .get();
+    }
+
+    private Station getStartStation() {
+        return sections.stream()
+                .map(Section::getUpStation)
+                .filter(upStation -> isStartStation(upStation))
+                .findFirst()
+                .get();
```

Comment:
Optional 을 get 하면 value 가 null 이면 `NoSuchElementException` 을 반환합니다.
```suggestion
                .orElseThrow(NoSuchElementException::new);
```

위와 같이 예외발생에 대해 알려주거나 혹은 커스텀한 예외를 만들어서 던저주면 좋습니다 😄